### PR TITLE
chore: bump vitest and add @vitest/coverage-c8

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
**Please describe the changes this PR makes and why it should be merged:**
Bumped vitest to 0.22.0 and added `@vitest/coverage-c8`
ref: https://github.com/vitest-dev/vitest/releases/tag/v0.22.0

**Status and versioning classification:**

- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: build.config.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export function createUnbuildConfig({
 	return defineBuildConfig({
 		entries,
 		clean: true,
+		failOnWarn: false,
```
Comment (ckohen) [inline build.config.ts:40]: This seems unrelated to this change?

**Review by ckohen — CHANGES_REQUESTED** [review]
While not strictly necessary since c8 is default, maybe we should specify for clarity https://vitest.dev/guide/coverage.html#coverage-providers

**Review by iCrawl — APPROVED** [review]
